src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/TextMapPropagator.cs

```diff
@@ -0,0 +1,139 @@
+
+        /// <summary>
+        /// Extracts the baggage key-value pairs list from an incoming request represented by the carrier. For example, from the headers of an HTTP request.
+        /// </summary>
+        /// <param name="carrier">Carrier is the medium used by the propagators to read values from.</param>
+        /// <param name="getter">The callback will be invoked to get the propagation baggage list from carrier.</param>
+        /// <returns>Returns the extracted key-value pair list from teh carrier.</returns>
+        public abstract IEnumerable<KeyValuePair<string, string?>>? ExtractBaggage(object carrier, PropagatorGetterCallback getter);
+
+        /// <summary>
+        /// Get or set the process wide propagator object which used as the current selected propagator.
+        /// </summary>
+        public static TextMapPropagator Current
+        {
+            get
+            {
+                Debug.Assert(s_current is not null);
+                return s_current;
+            }
+
+            set
+            {
+                if (value is not null)
+                {
+                    s_current = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// returns the default propagator object which Current property will be initialized with.
+        /// </summary>
+        public static TextMapPropagator CreateDefaultPropagator() => LegacyPropagator.Instance;
+
+        /// <summary>
+        /// returns the propagator which can propagate the trace context data using the root Activity and ignore any created child activities.
+        /// </summary>
+        public static TextMapPropagator CreatePassThroughPropagator() => PassThroughPropagator.Instance;
+
+        /// <summary>
+        /// returns the propagator which suppress injecting any data to teh carriers.
+        /// </summary>
+        public static TextMapPropagator CreateNoOutputPropagator() => NoOutputPropagator.Instance;
+
+        // internal stuff
+
+        internal static void InjectBaggage(object carrier, IEnumerable<KeyValuePair<string, string?>> baggage, PropagatorSetterCallback setter, bool injectAsW3C = false)
+        {
+            using (IEnumerator<KeyValuePair<string, string?>> e = baggage.GetEnumerator())
+            {
+                if (e.MoveNext())
+                {
+                    StringBuilder baggageList = new StringBuilder();
+
+                    KeyValuePair<string, string?> item = e.Current;
+                    baggageList.Append(WebUtility.UrlEncode(item.Key)).Append('=').Append(WebUtility.UrlEncode(item.Value));
+
+                    while (e.MoveNext())
+                    {
+                        item = e.Current;
+                        baggageList.Append(Comma).Append(WebUtility.UrlEncode(item.Key)).Append('=').Append(WebUtility.UrlEncode(item.Value));
```

Comment:
This should apppend "comma space" instead of comma to match HttpClient's behavior.

This loop could alternatively be written as `do while` with `baggageList.Length -= 2` at the end to avoid the duplication.

Reply:
oh, I forgot about this space thing. I'll fix it.